lib/datadog/tracing/span_operation.rb

```diff
@@ -83,12 +82,6 @@ def initialize(
         # Set tags if provided.
         set_tags(tags) if tags
 
-        # Only set parent if explicitly provided.
-        # We don't want it to override context-derived
-        # IDs if it's a distributed trace w/o a parent span.
-        parent = child_of
-        self.parent = parent if parent
```

Comment:
I tried to completely remove `#parent` as well, but it's still used for `service_entry` calculation.